source/common/stats/thread_local_store.h

```diff
@@ -225,6 +225,27 @@ class ThreadLocalStoreImpl : Logger::Loggable<Logger::Id::stats>, public StoreRo
     Histogram& histogram(const std::string& name) override;
     Histogram& tlsHistogram(const std::string& name, ParentHistogramImpl& parent) override;
 
+    template <class StatType>
+    using MakeStatFn = std::shared_ptr<StatType> (StatDataAllocator::*)(
```

Comment:
nit: This type of function pointer syntax took me by surprise and I suspect most people are not familiar with it. Is there any more readable way of doing this? Potentially a normal lambda that is passed an instance of StatDataAllocator? This is not a big deal, just throwing it out there.

Reply:
I can use std::function but I would have to make static method intermediates which I think is less readable.

In the past I have added macros for defining and calling pointer to member functions, which helps a bit I think.  That's my preferred option.  WDYT?

There is also std::invoke, which I had never used before. But I wasn't about to get that to with and I didn't think it was all that readable either.

Reply:
I can use std::function but I would have to make static method intermediates which I think is less readable.

In the past I have added macros for defining and calling pointer to member functions, which helps a bit I think.  That's my preferred option.  WDYT?

There is also std::invoke, which I had never used before. But I wasn't about to get that to with and I didn't think it was all that readable either.

Reply:
If you're going to use pointer to member function, I think macros are the preferred method for readability.

Why would you need static intermediates to use std::function? Can't you capture 'this' in the closure?

Reply:
You're right; I don't need a static intermediary, but I think the way the code is structured I do need an intermediate closure, because I'm not starting with 'this', I'm starting with two alternate allocators. I do have that working -- it's a little messy IMO -- and I'll push that for reference, and then I'll push the version where I use pointer to member functions, but via macros, and I'm fine with either of those.